diode-server/ingester/component.go

```diff
@@ -166,18 +168,20 @@ func (c *Component) Ingest(ctx context.Context, in *diodepb.IngestRequest) (*dio
 	}
 	ctx = telemetry.ContextWithMetricAttributes(ctx, attrs...)
 
-	if err := c.redisStreamClient.XAdd(ctx, &redis.XAddArgs{
+	err = c.redisStreamClient.XAdd(ctx, &redis.XAddArgs{
 		Stream: streamID,
 		Values: msg,
-	}).Err(); err != nil {
+	}).Err()
+	if err != nil {
 		c.metrics.RecordIngestRequest(ctx, false)
 		c.logger.Error("failed to add element to the stream", "error", err, "streamID", streamID, "value", msg)
-	} else {
-		entityCount := int64(len(in.GetEntities()))
-		c.metrics.RecordIngestRequest(ctx, true)
-		c.metrics.RecordIngestEntities(ctx, entityCount)
+		return nil, status.Error(codes.Internal, "")
```

Comment:
Perhaps we should use a code reserved for gRPC users? Like `ABORTED`?

https://grpc.io/docs/guides/status-codes/#full-list-of-status-codes

I don't think it matters much. Perhaps `INTERNAL` is the best for this use case.